docusaurus/docs/migration-guides/update-from-grafana-versions/v9.x-v10.x.md

```diff
@@ -37,3 +37,65 @@ Most code targeting 9.x will continue to work without any issues. An exception i
 When writing plugins that should run on 9.x, continue to use the Vector interfaces. In this case, when targeting versions 10+, you can now use simple arrays rather than wrapper classes.
 
 To make this transition seamless, we employed the Original JavaScript Sin™. That is, we [extended the native Array prototype](https://github.com/grafana/grafana/blob/v10.0.x/packages/grafana-data/src/types/vector.ts) with several Vector methods. We will atone and undo this in v11, when Vector interfaces and classes are removed.
+
+## Update to React Router v6
+
+Starting from Grafana 10 plugins should start updating to use the v6 version of `react-router`. Overall, `react-router` v6 aims to simplify route configuration and provide a more flexible and intuitive API for developers.
+
+**We strongly encourage developers to update their plugins to use the v6 version `react-router` as soon as possible, as the v5 version is going to be deprecated in the future.**
+
+[Official React Router v5 to v6 migration guide](https://reactrouter.com/en/main/upgrading/v5)
+
+### Update using `@grafana/create-plugin`
+
+Follow the steps below to start using React router v6 in your plugin.
+
+#### **1. Update the build related configuration:**
+
+Enable using `react-router@v6` by setting the following feature flag in `<project-root>/.cprc.json`:
+
+```json
+{
+  "features": {
+    "useReactRouterV6": true
+  }
+}
+```
+
+Now update the build configuration using the create-plugin tool:
+
+```bash
+npx @grafana/create-plugin@latest update
+```
+
+After updating the build config you probably also need to update some things in your code. Please follow the steps below:
+
+#### **2. Use `<Routes>` instead of `<Switch>`**
+
+```typescript
+// Using <Routes> instead of <Switch> in `react-router` v6
+import { Routes } from 'react-router-dom';
+
+// ...
+
+return (
+  <Routes>
+    <Route path="/" element={<Home />} />
+  </Routes>
+);
+```
+
+#### **3. Remove the `exact` prop from `<Route>` components**
+
+```typescript
+return (
+  <Routes>
+    {/* BAD (Until v5) */}
+    <Route exact path="/" element={<Home />} />
+
+    {/* GOOD (From v6) */}
+    {/* (Routes are "exact" by default, you need to use the "*" to match sub-routes) */}
+    <Route path="/" element={<Home />} />
+  </Routes>
+);
+```
```

Comment:
I'd add a step 4 here repeating the link to the migration guide, 

```
#### **4. Follow the migration guide for more in-depth change**
....
```

Reply:
Hmm, doesn't sound like a bad idea. Will update the PR. 👍

Reply:
Fixed in https://github.com/grafana/plugin-tools/pull/718/commits/21f6bd6ca0713734681a195146f827330cfb1f12.